python/tank/authentication/login_dialog.py

```diff
@@ -258,10 +346,13 @@ def _authenticate(self, error_label, site, login, password, auth_code=None):
             QtGui.QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)
             QtGui.QApplication.processEvents()
 
-            # try and authenticate
-            self._new_session_token = session_cache.generate_session_token(
-                site, login, password, self._http_proxy, auth_code
-            )
+            if session_token is None:
+                # try and authenticate
+                self._new_session_token = session_cache.generate_session_token(
+                    site, login, password, self._http_proxy, auth_code
+                )
+            else:
+                self._new_session_token = session_token
```

Comment:
Shouldn't you be doing the SSO authentication here instead? This way you'd have the benefit of having the wait cursors as well.